### ISSUE TITLE
grad is none for translation vector and quaternion

### ISSUE THREAD

**Issue by haksorus** [NPU-YanChi/diff-gaussian-rasterization-for-gsslam#1]
Hello!
Just tried your implementation with pose backward estimation backward and now im facing some troubles with gradient propagation - **cam_q_w2c and cam_t_w2c values has none grad after backward**.
Btw i create them from numpy with cast to float 32 and using cuda device with requires_grad=True

```python
cam_q_w2c = torch.tensor(Rotation.from_matrix(R).as_quat(), device='cuda:0', requires_grad=True).float()
cam_t_w2c = torch.tensor(T, device='cuda', requires_grad=True).float()
                
cam_info = CameraInfo(uid=i, R=R, T=T, cam_q_w2c=cam_q_w2c, cam_t_w2c=cam_t_w2c , FovY=FovY, FovX=FovX, cx=cx, cy=cy, fx=K_train, fy=K_train, image=image, image_path=image_path, image_name=image_name, width=image.size[0], height=image.size[1])
```
Rasterizer settings and function calling looks like this:
```python
raster_settings = GaussianRasterizationSettings(
        image_height=int(viewpoint_camera.image_height),
        image_width=int(viewpoint_camera.image_width),
        tanfovx=tanfovx,
        tanfovy=tanfovy,
        bg=bg_color,
        scale_modifier=scaling_modifier,
        viewmatrix=viewmatrix,
        projmatrix=viewpoint_camera.full_proj_transform,
        sh_degree=pc.active_sh_degree,
        campos=viewpoint_camera.camera_center,
        prefiltered=False,
        mapping_mode=False,
        tracking_mode=True,
        debug=False
        )
```

```python
# Rasterize visible Gaussians to image, obtain their radii (on screen). 
rendered_image, radii, rendered_depth, rendered_alpha, render_depth_var = rasterizer(
        means3D = means3D,
        means2D = means2D,
        shs = shs,
        colors_precomp = colors_precomp, 
        opacities = opacity,
        scales = scales,
        rotations = rotations,
        cov3D_precomp = cov3D_precomp,
        cam_q_w2c = viewpoint_camera.cam_q_w2c,
        cam_t_w2c = viewpoint_camera.cam_t_w2c)
```

Am I doing something wrong?

Thank you.

**Comment by yanchi-3dv** [comment]
Have replied by email

**Comment by while0l1** [comment]
I'm having the same problem, could you share the discussion here?

**Comment by Lily-Git-hub** [comment]
Hi, I had the same problem. Would you like to share the solution please? Thank you!

**Comment by Lily-Git-hub** [comment]
@NPU-YanChi @haksorus 
Hi, I had the same problem. Would you like to share the solution please? Thank you!

**Comment by yanchi-3dv** [comment]
I use this function to get t&q with nn.parameter.
![image](https://github.com/NPU-YanChi/diff-gaussian-rasterization-for-gsslam/assets/96414864/c538a52f-6907-45fd-9c84-3a46ccc9ef36)
And this is the tracking_render code(you can ignore coarse2fine here).
![image](https://github.com/NPU-YanChi/diff-gaussian-rasterization-for-gsslam/assets/96414864/bf3ad030-3318-462c-a262-743064bb9bce)
I think maybe two attempts that you can have a try:
1. Set mapping_mode to True
2. In each iters, I will update cur_q_w2c and cur_t_w2c and  in function “self.dataset.get_raster_settings” will update view_matric and proj_matric.

The grad is from loss -> cur_q_2c, cur_cur_t_w2c -> cur_trans, cur_rot.

Hope it is useful for you.

**Comment by JFcy** [comment]
Hello, I also want to reproduce the experiments from this project, but I am having trouble understanding the contents of the README. I'm not sure how to run or use the project. Could you please provide some help?
